rfcs/toast-component.md

```diff
@@ -0,0 +1,116 @@
+              console.log('Action button clicked');
+            }}>
+              Take action
+            </button>
+          </div>
+        );
+      }}
+      </Toast>
+    </React.Fragment>
+  );
+}
+```
+
+## Exports
+
+* `Toast`
+* `StyledRoot`
+* `StyledBody`
+* `StyledCloseButton`
+* `KIND`
+* `PLACEMENT`
+
+## `Toast` API
+
+* `children: React.node | ({dismiss}) => React.node` - Required
+  Toast notification content. The children-as-function receives a `dissmiss` method that can be called to dismiss the notification and can be used as a handler for an action inside the toast content
+* `closable: boolean` - Optional. Defaults to true
+  When set to true a close button is displayed and the notification can be dismissed by a user
+* `kind: 'info' | 'success' | 'warning' | 'error'` - Optional. Defaults to 'info'
+  Defines the type of notification
+* `onClose: function` - Optional
+  A callback function called when a notification is dismissed
+* `overrides: {Root, Body, CloseButton}` - Optional
+  Overrides for presentational components. See "Presentational Components Props API" below.
+  * `[ComponentName]: ReactComponent | {props: {}, style: {}, component: ReactComponent}` - Optional
+* `placement: 'inline' | 'topLeft' | 'top' | 'topRight' | 'bottomRight' | 'bottom' | 'bottomLeft'` - Optional. Defaults to 'top'
+  Position of a toast notification relative to the browser window. If set to 'inline' a notitication is positioned according to a normal flow of the document
+* `autoHideDuration: number` - Optional. Defaults to 0
+  The number of milliseconds to wait before automatically dismissing a notification. This behavior is disabled when the value is set to 0
+
+## Presentational components props API
+
+These properties are passed to every presentational (styled) component that is exported:
+
+* `$kind: type`
+* `$theme: theme`
+
+## KIND Constant
+
+* `info` - Generally used for messages with an informational context
+* `success` - Generally used as a confirmation of a succesful action or operation  
+* `warning` - Generally used for messages with an warning context
+* `error` - Generally used as a notification of an error happened as a result of an action or operation
+
+## PLACEMENT Constant
+
+* `inline` - positioned according to a normal flow of the document
+* `topLeft` - positioned relative to the browser window
+* `top` - positioned relative to the browser window
+* `topRight` - positioned relative to the browser window
+* `bottomRight` - positioned relative to the browser window
+* `bottom` - positioned relative to the browser window
+* `bottomLeft` - positioned relative to the browser window
+
+## Dependencies
+
+Does this component depend on any 3rd party packages or other internal components?
+
+## Accessibility
+
+* Toast notification container element will have `role="alert"` set on it
+* When multiple alerts are displayed and positioned relative to the browser window they won't overlap but displayed in a column way
```

Comment:
do we have a max number of Toast that can be stacked up on each other? what would happen if there would be more than the screen can fit?

Reply:
i think toasts should be controlled by some management tooling, which would avoid them to overlap each other, be hidden or outside of screen

Reply:
That's a good question. I looked through some examples, there are cases where everything that overflows the browser window (usually older/earlier notifications) is hidden. I think scrolling the notifications list would be an option. Will add this case to this section to consider for development.